Source/Extensions/Blazorise.Components/SelectList.razor.cs

```diff
@@ -147,6 +235,13 @@ public Task Focus( bool scrollToElement = true )
         /// </summary>
         [Parameter] public RenderFragment ChildContent { get; set; }
 
+
+        /// <summary>
+        /// Returns true if SelectedItem property is supported.
+        /// </summary>
+        public bool SelectedItemSupported =>
+            ValueField == null && Data is IList && typeof(TValue) == typeof(int);
```

Comment:
Why support only `typeof(int)`? I think practically any `TValue` type should be indexed.

Reply:
In a IList, the index is always int. A standard collection supporting other types as index would be a dictionary which is not assignable from IEnumerable<TItem>. We would need to move to object for Data (btw as WPF/UWP does with ItemSource)